.github/workflows/ci-on-pr-labels.yaml

```diff
@@ -0,0 +1,40 @@
+# Copyright (c) 2023 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+name: Kata Containers PR CI on label change
+on:
+  pull_request:
+    types:
+      - labeled
+      - unlabeled
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  skip:
+    uses: ./.github/workflows/skip-pr-check.yaml
+    with:
+      pr: ${{ github.event.pull_request.number }}
+    secrets: inherit
```

Comment:
I'm not confident this will work.
When we're dealing with GitHub actions targetting `pull_request`, no secrets are passed down to workflows, apart from `GITHUB_TOKEN`, and this is documented here: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflows-in-forked-repositories

Reply:
Indeed that seems to be case

Reply:
Actually, I think the documentation is a bit misleading. It says "secrets are not passed to the runner when a workflow is triggered from a forked repository." That means you fork the repo and trigger the workflow on the fork. That's not what happens during PRs. The PR triggers workflows in the repo you are trying to merge to, not the fork. So this limitation doesn't apply AFAICT.

Reply:
See description at https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull-request-events-for-forked-repositories: "No pull request events occur on the forked repository."

Reply:
To double-check I've setup an experiment at https://github.com/tklengyel/pr_secret_test/pulls to check whether a PR has access to secrets when triggering a workflow. It doesn't and no matter whether the workflow was triggered directly by the PR event or down the line via workflow calls.

That means the workflows already in place with a PR trigger in kata-containers that use any secret token other then the default GITHUB_TOKEN are broken.

Reply:
Wow, OK. There is a difference between `pull_request` and `pull_request_target` triggers. With `pull_request_target` the workflow will have access to the secrets. What a pain :)